frontend/src/metabase/dashboard/actions/auto-wire-parameters/actions.ts

```diff
@@ -123,46 +123,43 @@ export function autoWireParametersToNewCard({
       return;
     }
 
-    const dashcardMappingOptions = getParameterMappingOptions(
+    const targetQuestion =
       questions[targetDashcard.card.id] ??
-        new Question(targetDashcard.card, metadata),
+      new Question(targetDashcard.card, metadata);
+
+    const dashcardMappingOptions = getParameterMappingOptions(
+      targetQuestion,
       null,
       targetDashcard.card,
       targetDashcard,
     );
 
-    const targetQuestion =
-      questions[targetDashcard.card.id] ??
-      new Question(targetDashcard.card, metadata);
-
     const parametersToAutoApply = [];
     const processedParameterIds = new Set();
 
-    for (const opt of dashcardMappingOptions) {
-      for (const [dashcard, question] of dashcardWithQuestions) {
-        const param = dashcard.parameter_mappings?.find(mapping =>
-          compareMappingOptionTargets(
-            mapping.target,
-            opt.target,
-            question,
-            targetQuestion,
-          ),
+    for (const [dashcard] of dashcardWithQuestions) {
```

Comment:
it looks like we can get rid of `dashcardWithQuestions` and use `for (const dashcard of dashcards) {`

Reply:
Agree, fixed